components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java

```diff
@@ -307,6 +329,59 @@ private String getResourceName(String type, String name, String locale) {
         return resourceName;
     }
 
+    /**
+     * Trigger pre add branding preference events.
+     *
+     * @param brandingPreference Branding Preference.
+     * @param tenantDomain       Tenant domain.
+     */
+    private void triggerPreAddBrandingPreferenceEvents(BrandingPreference brandingPreference, String tenantDomain)
+            throws BrandingPreferenceMgtException {
+
+        Map<String, Object> eventProperties = new HashMap<>();
+        eventProperties.put(BRANDING_PREFERENCE, brandingPreference);
+        eventProperties.put(TENANT_DOMAIN, tenantDomain);
+        Event brandingPreferenceEvent = new Event(PRE_ADD_BRANDING_PREFERENCE, eventProperties);
```

Comment:
Shall we add this event as well to the PR description?

Reply:
Updated